features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/metric/SessionEndedMetric.kt

```diff
@@ -0,0 +1,215 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.metric
+
+import com.datadog.android.core.metrics.PerformanceMetric
+import com.datadog.android.rum.internal.domain.scope.RumSessionScope
+import com.datadog.android.rum.internal.domain.scope.RumViewManagerScope
+import com.datadog.android.rum.model.ViewEvent
+import java.util.concurrent.TimeUnit
+
+/**
+ * Metric for rum session ended event.
+ */
+@Suppress("TooManyFunctions")
+internal class SessionEndedMetric(
+    private val sessionId: String,
+    private val startReason: RumSessionScope.StartReason
+) {
+
+    private val trackedViewsById = mutableMapOf<String, TrackedView>()
+
+    private val errorKindFrequencies = mutableMapOf<String, Int>()
+
+    private var firstTrackedView: TrackedView? = null
+    private var lastTrackedView: TrackedView? = null
+    private var wasStopped: Boolean = false
```

Comment:
minor: it is better to use present tense here

```suggestion
    private var isStopped: Boolean = false
```

Reply:
The attribute name in the json is "was_stopped", is it better to align the property name with this?

Reply:
I think we can keep class property name different from the attribute name. But not a big deal tbh.

Reply:
OK, since that I prefer keeping it, so that it makes property refer the attribute.